src/utils/deserialize.ts

```diff
@@ -0,0 +1,261 @@
+import { PublicKey } from '@solana/web3.js';
+import {
+  BinaryReader, Schema, BorshError, BinaryWriter,
+} from 'borsh';
+
+(BinaryReader.prototype as any).readPubkey = function () {
+  const reader = this as unknown as BinaryReader;
+  const array = reader.readFixedArray(32);
+  return new PublicKey(array);
+};
+
+(BinaryWriter.prototype as any).writePubkey = function (value: PublicKey) {
+  const writer = this as unknown as BinaryWriter;
+  writer.writeFixedArray(value.toBuffer());
+};
+
+export enum VoteTypeKind {
+  SingleChoice = 0,
+  MultiChoice = 1,
+}
+
+export class VoteType {
+  type: VoteTypeKind;
+
+  choiceCount: number | undefined;
+
+  constructor(args: { type: VoteTypeKind; choiceCount: number | undefined }) {
+    this.type = args.type;
+    this.choiceCount = args.choiceCount;
+  }
+
+  static SINGLE_CHOICE = new VoteType({
+    type: VoteTypeKind.SingleChoice,
+    choiceCount: undefined,
+  });
+
+  static MULTI_CHOICE = (choiceCount: number) => new VoteType({
+    type: VoteTypeKind.MultiChoice,
+    choiceCount,
+  });
+
+  isSingleChoice() {
+    return this.type === VoteTypeKind.SingleChoice;
+  }
+}
+
+(BinaryReader.prototype as any).readVoteType = function () {
+  const reader = this as unknown as BinaryReader;
+  const value = reader.buf.readUInt8(reader.offset);
+  reader.offset += 1;
+
+  if (value === VoteTypeKind.SingleChoice) {
+    return VoteType.SINGLE_CHOICE;
+  }
+
+  const choiceCount = reader.buf.readUInt16LE(reader.offset);
+  return VoteType.MULTI_CHOICE(choiceCount);
+};
+
+export enum VoteThresholdType {
+  // Approval Quorum
+  YesVotePercentage = 0,
+  // Not supported in the current version
+  QuorumPercentage = 1,
+  // Supported for VERSION >= 3
+  Disabled = 2,
+}
+
+export class VoteThreshold {
+  type: VoteThresholdType;
+
+  value: number | undefined;
+
+  constructor(args: { type: VoteThresholdType; value?: number | undefined }) {
+    this.type = args.type;
+    this.value = args.value;
+  }
+}
+
+(BinaryReader.prototype as any).readVoteThreshold = function () {
+  const reader = this as unknown as BinaryReader;
+
+  // Read VoteThreshold and advance the reader by 1
+  const type = reader.buf.readUInt8(reader.offset);
+  reader.offset += 1;
+
+  // Read VoteThresholds with u8 value
+  if (
+    type === VoteThresholdType.YesVotePercentage
+    || type === VoteThresholdType.QuorumPercentage
+  ) {
+    const percentage = reader.buf.readUInt8(reader.offset);
+    reader.offset += 1;
+    return new VoteThreshold({ type, value: percentage });
+  }
+
+  // Read VoteThresholds without value
+  if (type === VoteThresholdType.Disabled) {
+    return new VoteThreshold({ type, value: undefined });
+  }
+
+  throw new Error(`VoteThresholdType ${type} is not supported`);
+};
+
+function capitalizeFirstLetter(string: string): string {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+function deserializeField(
```

Comment:
Where is the rest of this used?

Reply:
What rest?

This is coped from the governance oyster repo. Didnt directly include that sdk because of the build issue linked in our code in the ui.

Reply:
Answers this thanks